src/test/java/com/iota/iri/service/snapshot/impl/LocalSnapshotManagerImplTest.java

```diff
@@ -0,0 +1,156 @@
+package com.iota.iri.service.snapshot.impl;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.exceptions.base.MockitoAssertionError;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import com.iota.iri.conf.SnapshotConfig;
+import com.iota.iri.service.milestone.LatestMilestoneTracker;
+import com.iota.iri.service.snapshot.SnapshotException;
+import com.iota.iri.service.snapshot.SnapshotProvider;
+import com.iota.iri.service.snapshot.SnapshotService;
+import com.iota.iri.service.transactionpruning.TransactionPruner;
+import com.iota.iri.utils.thread.ThreadUtils;
+
+public class LocalSnapshotManagerImplTest {
+    
+    private static final int BUFFER = LocalSnapshotManagerImpl.LOCAL_SNAPSHOT_SYNC_BUFFER;
+    
+    private static final int DELAY_SYNC = 5;
+    private static final int DELAY_UNSYNC = 1;
+    private static final int SNAPSHOT_DEPTH = 5;
+
+    @Rule 
+    public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    @Mock
+    private static SnapshotConfig config;
+    
+    
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    SnapshotProvider snapshotProvider;
+    
+    @Mock
+    SnapshotService snapshotService;
+    
+    @Mock
+    TransactionPruner transactionPruner;
+    
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    LatestMilestoneTracker milestoneTracker;
+    
+    private LocalSnapshotManagerImpl lsManager;
+
+    @Before
+    public void setUp() throws Exception {
+        this.lsManager = new LocalSnapshotManagerImpl();
+        
+        lsManager.init(snapshotProvider, snapshotService, transactionPruner, config);
+        when(snapshotProvider.getLatestSnapshot().getIndex()).thenReturn(-5, -1, 10, 998, 999, 1999, 2000);
```

Comment:
Hmm fair point, I just figured that if we ever break this its a good reminder to think things through. As it could maybe break something else somewhere. (not in IRI)